test/controllers/IllnessStartDateController.test.ts

```diff
@@ -0,0 +1,208 @@
+                    illness: {
+                        illnessStart: '2020-01-01T00:00:00.000Z'
+                    }
+                }
+            } as Appeal;
+
+            const app = createApp({appeal: appealWithReasons});
+            await request(app).get(ILLNESS_START_DATE_PAGE_URI)
+                .expect(response => {
+                    expect(response.status).to.be.equal(OK);
+                    expect(response.text).to.contain(pageHeading)
+                        .and.to.contain('id="start-day" name="day" type="text" value="1"')
+                        .and.to.contain('id="start-month" name="month" type="text" value="1"')
+                        .and.to.contain('id="start-year" name="year" type="text" value="2020"');
+                });
+        });
+
+        it('should redirect to entry page when illness reason feature is disabled', async () => {
+
+            process.env.ILLNESS_REASON_FEATURE_ENABLED = '0';
+
+            const app = createApp({appeal});
+            await request(app).get(ILLNESS_START_DATE_PAGE_URI)
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ENTRY_PAGE_URI);
+                });
+        });
+    });
+
+    describe('POST request', () => {
+
+        beforeEach(() => {
+            process.env.ILLNESS_REASON_FEATURE_ENABLED = '1';
+        });
+
+        it('should redirect to illness start date page when posting a valid date', async () => {
+            const app = createApp({appeal});
+            await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                .send({day: '01', month: '01', year: '2020'})
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ILLNESS_START_DATE_PAGE_URI);
+                });
+        });
+
+        it('should return 422 response with rendered error messages when empty start date was submitted', async () => {
+            const app = createApp({appeal});
+            await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                .send({})
+                .expect(response => {
+                    expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                    expect(response.text).to.include(pageHeading)
+                        .and.to.include(errorSummaryHeading)
+                        .and.to.include(invalidStartDayErrorMessage)
+                        .and.to.include(invalidStartMonthErrorMessage)
+                        .and.to.include(invalidStartYearErrorMessage)
+                        .and.to.not.include(invalidDateErrorMessage)
+                        .and.to.not.include(invalidDateFutureErrorMessage);
+                });
+        });
+
+        it('should return 422 response with rendered error messages when partial start date was submitted',
```

Comment:
For the future, I think it is bit to much to check validation in these integration tests. I'd be happy to say it worked after seeing all validation messages after submitting empty form.

You can live it this time but generally not required in the future.